Mathlib/Data/MLList/BestFirst.lean

```diff
@@ -29,37 +31,198 @@ Options:
 
 set_option autoImplicit true
 
+open Std EstimatorData Estimator Set
 
-variable {α : Type u} [Monad m] [Alternative m] [Ord α]
+/-!
+We begin by defining a best-first queue of `MLList`s.
+This is a somewhat baroque data structure designed for the application in this file
+(and in particularly for the implementation of `rewrite_search`).
+If someone would like to generalize appropriately that would be great.
+
+We want to maintain a priority queue of `MLList m β`, each indexed by some `a : α` with a priority.
+To complicate matters, the priorities might be expensive to calculate,
+so we instead keep track of a lower bound for each such `a : α`.
```

Comment:
In this context of best-first search, I initially interpreted lower priorities to mean worse, but here they are better, right? (This order on priority goes against the conventions in Lean where lower priorities are visited later, maybe we should replace "priority" with another word?)
```suggestion
so we instead keep track of a lower bound (where less is better) for each such `a : α`.
```